### PR TITLE
Patch 1

### PR DESCRIPTION
**Pull request by shuvaday**
Updated with review comments.

### CODE REVIEW

File: Python/omevv/v1/scripts/CreateRepoProfile.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
+import argparse
+import sys
+import time
+import warnings
+import base64
+import constants
+from webbrowser import get
+import re
+from omevv.v1.omevv_apis_client.models import Credential, ShareCredential
```
Comment (pavithra-mahadev) [inline Python/omevv/v1/scripts/CreateRepoProfile.py]: Why do we need ShareCredential?

File: Python/omevv/v1/scripts/CreateRepoProfile.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    parser.add_argument('-domain', help='domain of share path', required=False);
+
+    args = vars(parser.parse_args())
+    ome_ip = args['ip'];
+    omevv_user_name = args['omevv_user'];
+    omevv_cred = args['omevv_pswd'];
+    uuid = args['uuid'];
+
+    if args['protocol'] == "CIFS" and args['spcred'] == "":
+        print('credential missing for sharedpath for CISF protocol')
```
Comment (pavithra-mahadev) [inline Python/omevv/v1/scripts/CreateRepoProfile.py]: typo - CIFS

File: Python/omevv/v1/scripts/CreateRepoProfile.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    if args['profilename'] == "" and args['profilename'] is None:
+        print('Profilename is a required parameter.Please pass a value for the same')
+    obj = CreateRepo(ome_ip,omevv_user_name,omevv_cred,uuid)
+    if (obj.validate_path(args["protocol"], args["sharepath"]) is not True):
+        print('Please pass valid sharedpath location for the provided protocl type')
```
Comment (pavithra-mahadev) [inline Python/omevv/v1/scripts/CreateRepoProfile.py]: typo - protocol

File: Python/omevv/v1/scripts/CreateRepoProfile.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    );
+    parser.add_argument('-ip', help='OMEVV IP address', required=True);
+    parser.add_argument('-omevv_user', help='OMEVV username', required=True);
+    parser.add_argument('-omevv_pswd', help='OMEVV password', required=True);
+    parser.add_argument('-profilename', help='repository profile name', required=True);
+    parser.add_argument('-description', help='repository profile description', required=False);
+    parser.add_argument('-profileType', help='repository profile type', choices=['DRIVER', 'FIRMWARE'], required=True);
+    parser.add_argument('-sharepath', help='sharepath', required=True);
+    parser.add_argument('-spcred', help='sharepath credential', required=False);
+    parser.add_argument('-protocol', help='protocol type', choices=['CIFS', 'NFS', 'HTTP', 'HTTPS', 'UNAVAILABLE'],
```
Comment (pavithra-mahadev) [inline Python/omevv/v1/scripts/CreateRepoProfile.py:101]: can we give UNAVAILABLE?

**Review by pavithra-mahadev — CHANGES_REQUESTED** [review]

File: Python/omevv/v1/scripts/CreateRepoProfile.py (hunk 1 above)
Comment (shuvaday) [reply Python/omevv/v1/scripts/CreateRepoProfile.py]: for share path credentials for CIFS

File: Python/omevv/v1/scripts/CreateRepoProfile.py (hunk 4 above)
Comment (shuvaday) [reply Python/omevv/v1/scripts/CreateRepoProfile.py:101]: dnt understand

File: Python/omevv/v1/scripts/CreateRepoProfile.py (hunk 2 above)
Comment (shuvaday) [reply Python/omevv/v1/scripts/CreateRepoProfile.py]: done

File: Python/omevv/v1/scripts/CreateRepoProfile.py (hunk 3 above)
Comment (shuvaday) [reply Python/omevv/v1/scripts/CreateRepoProfile.py]: done

File: Python/omevv/v1/scripts/CreateRepoProfile.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+    def __init__(self,ome_ip,omevv_user,omevv_pswd,uuid,):
+        self.headers ={}
+        self.payload = {}
+        credential = Credential(username=omevv_user, password=omevv_pswd)
+        appendedcredential = credential.username + ":" + credential.password
+        basicAuth = "Basic %s" % base64.b64encode(appendedcredential.encode('utf-8')).decode()
+        self.headers["Authorization"] = basicAuth
+        self.headers["Content-Type"] = 'application/json'
+        self.headers['x_omivv-api-vcenter-identifier'] = uuid
```
Comment (pavithra-mahadev) [inline Python/omevv/v1/scripts/CreateRepoProfile.py]: use constants

File: Python/omevv/v1/scripts/CreateRepoProfile.py [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+import sys
+import time
+import warnings
+import base64
+import constants
+from webbrowser import get
+import re
+from omevv.v1.omevv_apis_client.models import Credential, ShareCredential
+from omevv.v1.omevv_apis_client.models import ManagedHost
+from omevv.v1.omevv_apis_client.models import ErrorObject,protocol_type,profile_type,share_credential
```
Comment (pavithra-mahadev) [inline Python/omevv/v1/scripts/CreateRepoProfile.py]: remove one share_credential

**Review by pavithra-mahadev — APPROVED** [review]

**Review by pavithra-mahadev — APPROVED** [review]

**Review by Abhishek14-Dell — APPROVED** [review]

**Review by prasadrao-dell — APPROVED** [review]

**Review by pavithra-mahadev — APPROVED** [review]

**Pull request merged**